src/plots/cartesian/axes.js

```diff
@@ -1261,7 +1261,8 @@ function syncTicks(ax) {
     return ticksOut;
 }
 
-function arrayTicks(ax) {
+function arrayTicks(ax, major) {
+    if (major === undefined) throw new Error("arrayTicks must specify ticktype")
```

Comment:
Do we really need this line?
After you added it in https://github.com/plotly/plotly.js/pull/6829/commits/9a6de432024296f3990ddd1250d7376fb801bf4e, some tests failed.

Reply:
That's good actually! 

I put that line in there to force failure if other calls relied on default behavior.

Jasmine is wonky on my local box so I didn't catch those failed tests before push.

Reply:
The arrayTicks will now behave in a _legacy_ manner if the caller isn't aware to use the major/minor switch.